analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd

```diff
@@ -0,0 +1,261 @@
+        # light guiding line representing 0 exposure
+        geom_hline(yintercept = 0, size = 0.15) + 
+        scale_color_manual(values = palette, 
+                           breaks = sort(names(palette))) +
+        stat_summary(color = "black", size = 0.3) + 
+        labs(x = "Tumor descriptor",
+             y = "TMB") +
+        theme_Publication() +
+        scale_y_continuous(limits = c(0, ylim)) +
+        theme(axis.text.x = element_text(angle = 90)))
+
+# Save the plot
+ggsave(filename = "TMB-unpaired-longitudinal.pdf",
+       path = plots_dir, 
+       width = 10, 
+       height = 5, 
+       device = "pdf", 
+       useDingbats = FALSE)
+```
+
+
+# All unpaired samples across cancer types
+
+```{r plot-cg-id-sum, fig.width = 25, fig.height = 20, fig.fullwidth = TRUE}
+print(ggplot(df_plot, aes(tumor_descriptor, log10_tmb, color = Timepoint)) + 
+        geom_jitter(width = 0.2, alpha = 0.5, size = 1.5) + 
+        # light guiding line representing 0 exposure
+        geom_hline(yintercept = 0, size = 0.15) + 
+        scale_color_manual(values = palette, 
+                           breaks = sort(names(palette))) +
+        stat_summary(color = "black", size = 0.3) + 
+        ggplot2::facet_wrap(~cg_id_sum, nrow = 8) +
+        labs(x = "Tumor descriptor",
+             y = "log10TMB") +
+        theme_Publication() +
+        scale_y_continuous(limits = c(0, log10_ylim)) +
+        theme(axis.text.x = element_text(angle = 90)))
+
+# Save the plot
+ggsave(filename = "TMB-unpaired-longitudinal-cg-id-sum.pdf", 
+       path = plots_dir, 
+       width = 25, 
+       height = 20, 
+       device = "pdf", 
+       useDingbats = FALSE)
+```
+
+# Summary statistics for Timepoints and cancer types
+Considering the inequalities in the sampling effort for each timepoint, let's explore whether the sampling size affects the observed patterns, e.g., indication of Progressive and Recurrence having higer TMB compared to Diagnosis for High-grade glioma patients.
+
+```{r summary-statistics, fig.width = 12, fig.height = 6, fig.fullwidth = TRUE}
+# Inspect some random rows of the data by groups
+set.seed(2023)
+df_plot_cgGFAC <- df_plot %>% 
+  arrange(tumor_descriptor_cgGFAC_n) 
+
+df_plot_cgGFAC %>% 
+  select(cgGFAC, tumor_descriptor_cgGFAC_n, log10_tmb) %>% 
+  sample_n_by(cgGFAC, tumor_descriptor_cgGFAC_n, log10_tmb, size = 1) 
+
+# Compute some summary statistics (mean and sd) by groups:
+stat.summary <- df_plot_cgGFAC %>%
+  group_by(cgGFAC, tumor_descriptor_cgGFAC_n) %>%
+  get_summary_stats(log10_tmb, type = "mean_sd") %>% 
+  write_tsv(file.path(results_dir, "stat_summary.tsv"))
+
+# Pairwise comparisons between time points at each group levels
+# Paired t-test is used because we have repeated measures by time
+stat.test <- df_plot_cgGFAC %>%
+  group_by(cgGFAC) %>%
+  pairwise_t_test(log10_tmb ~ tumor_descriptor_cgGFAC_n, 
+                  paired = FALSE, 
+                  p.adjust.method = "bonferroni") %>% 
```

Comment:
you might consider doing a `wilcox_paired_test()` instead, to avoid parametric assumptions of t tests. Also, is `paired = FALSE` correct here?

Reply:
That is a great point, thanks! This is now implemented. 
`paired = FALSE` was used here because the number of arguments per group (i.e., timepoint_cgGFAC_n) is uneven, e.g., `Atypical Teratoid Rhabdoid Tumor_Recurrence  (N=3)` vs `Atypical Teratoid Rhabdoid Tumor_Progressive  (N=10)`. 
`Paired = TRUE` could be used more in cases with before/after.